flake.nix

```diff
@@ -65,7 +70,14 @@
           # `nix develop`
           devShells.default = pkgs.mkShell
             {
-              buildInputs = commonArgs.buildInputs ++ [ pkgs.pkg-config pkgs.systemd ];
+              NIX_CFLAGS_LINK = "-fuse-ld=mold";
+              
+              buildInputs = with pkgs;[
+                mold
+                clang
```

Comment:
tbh I am not sure what to think about mold.
In general it looks promising and I know it is in stable release now, but I don't have a clue about the implications for the binaries produced.

Reply:
I only use it because it's a faster linker, nothing else

Reply:
Maybe this is just my ignorance of how linking actually works. 
As I understood this so far the linker is building the final binary, right? So isn't there a "risk" that different linkers result in different binaries and in the worst case - caused by a linker bug - result in a bug of the binary itself?

I just want to be sure we don't place a potential foot gun here, that might send us on a potentially nasty bug hunt.
Also (not that I have done that yet, but eventually might look into it, or someone else wants to use it) how well does mold integrate into IDE debuggers? AFAIK most of the IDEs rely on llvm for debugging rust. 

So far I only found comparisons of speed for lld vs mold, but if you could educate me on the topic and clear out my worries, I would be happy to go for mold.